src/components/BackOffice/hooks/useSelectContract.js

```diff
@@ -46,7 +46,8 @@ const useSelectContract = () => {
   }, [selectedContractor])
 
   const handleSelectContractor = (e) => {
-    const contractorName = e.target.value
+    // Ternary operator prevents the line below from erroring when "resetForm() calls handleSelectContractor(null)"
+    const contractorName = e ? e.target?.value : undefined
```

Comment:
Ternary operator prevented the line below from erroring when "resetForm() calls handleSelectContractor(null)"
After this change the form is reset correctly (manually tested)